ext/NonlinearSolveSIAMFANLEquationsExt.jl

```diff
@@ -2,51 +2,58 @@ module NonlinearSolveSIAMFANLEquationsExt
 
 using NonlinearSolve, SciMLBase
 using SIAMFANLEquations
-import ConcreteStructs: @concrete
 import UnPack: @unpack
 
-function SciMLBase.__solve(prob::NonlinearProblem, alg::SIAMFANLEquationsJL, args...; abstol = nothing,
-        reltol = nothing, alias_u0::Bool = false, maxiters = 1000, termination_condition = nothing, kwargs...)
-    @assert (termination_condition === nothing) || (termination_condition isa AbsNormTerminationMode) "SIAMFANLEquationsJL does not support termination conditions!"
+@inline function __siam_fanl_equations_retcode_mapping(sol)
+    if sol.errcode == 0
+        return ReturnCode.Success
+    elseif sol.errcode == 10
+        return ReturnCode.MaxIters
+    elseif sol.errcode == 1
+        return ReturnCode.Failure
+    elseif sol.errcode == -1
+        return ReturnCode.Default
+    end
+end
+
+# pseudo transient continuation has a fixed cost per iteration, iteration statistics are
+# not interesting here.
+@inline function __siam_fanl_equations_stats_mapping(method, sol)
+    method === :pseudotransient && return nothing
+    return SciMLBase.NLStats(sum(sol.stats.ifun), sum(sol.stats.ijac), 0, 0,
+        sum(sol.stats.iarm))
+end
 
-    @unpack method, show_trace, delta, linsolve = alg
+function SciMLBase.__solve(prob::NonlinearProblem, alg::SIAMFANLEquationsJL, args...;
+        abstol = nothing, reltol = nothing, alias_u0::Bool = false, maxiters = 1000,
+        termination_condition = nothing, show_trace::Val{ShT} = Val(false),
+        kwargs...) where {ShT}
+    @assert (termination_condition ===
+             nothing)||(termination_condition isa AbsNormTerminationMode) "SIAMFANLEquationsJL does not support termination conditions!"
+
+    @unpack method, delta, linsolve = alg
 
     iip = SciMLBase.isinplace(prob)
-    T = eltype(prob.u0)
 
-    atol = abstol === nothing ? real(oneunit(T)) * (eps(real(one(T))))^(4 // 5) : abstol
-    rtol = reltol === nothing ? real(oneunit(T)) * (eps(real(one(T))))^(4 // 5) : reltol
+    atol = NonlinearSolve.DEFAULT_TOLERANCE(abstol, eltype(prob.u0))
+    rtol = NonlinearSolve.DEFAULT_TOLERANCE(reltol, eltype(prob.u0))
 
     if prob.u0 isa Number
-        f! = if iip
-            function (u)
-                du = similar(u)
-                prob.f(du, u, prob.p)
-                return du
-            end
```

Comment:
This shouldn't originate in the first place.

Reply:
Yeah, the scalar case root finding doesn't need this